### PR TITLE
Added Country field to the School

### PR DESCRIPTION
**Pull request by owentanuwidjaja**

### CODE REVIEW

**Comment by Reimirno** [comment]
You should pr to our repo first. Additionally you should add/fix relevant tests.

**Comment by Reimirno** [comment]
Taken a brief look. Generally on the right track. A few things I noticed:

1. You probably need a data migration script to give old school entries a fallback country value, since now country becomes a non-nullable column. Otherwise this causes database inconsistency.
2. There is no database validation for the country column. We are only checking its presence. What needs to be done too is to validate the value against the collection of valid countries.
3. Please add and fix relevant tests.

**Comment by cycomachead** [comment]
Thanks for the comments. I do agree we should add a few test cases. There are a couple of things we should clean up before merging.

> You probably need a data migration script to give old school entries a fallback country value, since now country becomes a non-nullable column. Otherwise this causes database inconsistency.

I think this current migration allows null values, for now, that actually seems fine. 

Today, we have a bunch of schools with the state "International" -- we need to clean this up.

* For schools that have 'International' as the state, we should remove that, and set the state to the empty string, and I suppose the country as well. Then we can manually fix the schools...
* We should remove 'International' as a valid state. US state validations should only apple when the country is 'US'.
* We should, for now, allow blank states when the country is not the US. 

In the future:
* if the user changes the country to not the US, we should probably reset the list of states and allow an open ended text field
* We should update the function to set coordinates to use the country the user selects

File: app/views/schools/_form.html.erb [hunk 1]
```diff
@@ -6,6 +6,17 @@
       required: false, id: 'school_name' %>
   </div>
 
+  <div class="form-group row">
+  <div class='col-12'>
+    <%= f.label :country, "Country", class: "label-required" %>
+    <%= f.country_select(
+      :country,
+      { priority_countries: ['United States'], include_blank: "Select an option" },
```
Comment (cycomachead) [inline app/views/schools/_form.html.erb:14]: Can we add:

```
priority_countries_divider: "------"
```

**Comment by Reimirno** [comment]
Yes. the schema migration is not making the field non-nullable. My bad :)

File: app/views/schools/_form.html.erb (hunk 1 above)
Comment (owentanuwidjaja) [reply app/views/schools/_form.html.erb:24]: The United States is already separated by the "--------" divider

**Comment by owentanuwidjaja** [comment]
I have already added the test cases based on the feature, I added.

For some reason, it won't let me create a new school when I leave the state empty for Non-US countries even though I have made it so that non-US countries isn't required to fill out the state option.  I created a javascript that tracks user input for the country and if it is not the US it does not require a state but there is always an error saying that "State is not included in the list". I also tried to completely remove the validation for the State and it still shows up with this error "State is not included in the list"

So I made it so that whenever US is chosen as the Country, it will populate the state options with the 50 states and whenever a non-US country is chosen only option is "International". Is this okay for now?

**Pull request closed without merging**